### PR TITLE
fix(netlify-edge): update manifest pattern and output format

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

https://github.com/nuxt/framework/issues/7592, https://github.com/nuxt/framework/issues/5509

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Currently nitro emits an invalid manifest for netlify edge. This updates the pattern to match the current netlify edge requirements.

**Note**: Once fixed, there is another issue I'm experiencing with netlify edge and will be investigating that next.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/presets/netlify.ts [hunk 1]
```diff
@@ -64,7 +64,8 @@ export const netlifyEdge = defineNitroPreset({
   },
   rollupConfig: {
     output: {
-      entryFileNames: 'server.js'
+      entryFileNames: 'server.js',
+      format: 'esm'
```
Comment (pi0) [inline src/presets/netlify.ts:68]: Why is this required?

Comment (danielroe) [reply src/presets/netlify.ts:68]: because otherwise it does not export anything but outputs an IIFE.
